lib/src/room.dart

```diff
@@ -618,6 +618,14 @@ class KurentoRoom extends ChangeNotifier implements Room {
           // Agent-initiated end message, but the impact of that is low (the Explorer can end the call themselves).
           _getServiceRequestStatusTimer?.cancel();
 
+          // HACK: In some cases, the video track isn't fully created at the time we join the room. This causes Agent
+          // Dashboard to behave as if the video is muted when it really isn't. So, if we have a video track and we're
+          // not muted, we want to make sure SFUConnection is transmitting that video.
+          final bool shouldTransmitVideo = _isVideoMuted == false && _hasVideoTrack;
+          if (shouldTransmitVideo && _connectionByTrackId[_localTrackId!]?.ownsVideoTrack != true) {
```

Comment:
I'm confused by something. I'm guessing the issue lies in this code in `_updateParticipantStatus`:

```dart
    bool isActiveVideoMuted = _isPresenting
        ? _isPresentationMuted
        : _isVideoMuted || !_hasVideoTrack; // Dash displays a call with no video the same as a call with muted video.
```

Specifically, if `_localStream?.getVideoTracks()` is missing or empty, we will send `PRIVACY_AUDIO_ONLY` to Dash.

But this fix also relies on `_hasVideoTrack`. So how does it work? If it was `true`, wouldn't we have sent a different status to Dash?

Reply:
No because these two calls are made at two very different time in our process and we expect, in this specific case, that the video track might now be available.